lib/command/src/Obelisk/Command/Run.hs

```diff
@@ -88,21 +86,44 @@ run = do
       , "Frontend.frontend"
       ]
 
+-- | Nix syntax requires relative paths to be prefixed by @./@ or
+-- @../@. This will make a 'FilePath' can be embedded in a Nix
+-- expression.
+toNixPath :: FilePath -> FilePath
+toNixPath root | "/" `isInfixOf` root = root
+               | otherwise = "./" <> root
+
 runRepl :: MonadObelisk m => m ()
 runRepl = do
-  pkgs <- getLocalPkgs
+  pkgs <- withProjectRoot "." getLocalPkgs
   withGhciScript pkgs $ \dotGhciPath -> do
     runGhciRepl dotGhciPath
 
 runWatch :: MonadObelisk m => m ()
 runWatch = do
-  pkgs <- getLocalPkgs
+  pkgs <- withProjectRoot "." getLocalPkgs
   withGhciScript pkgs $ \dotGhciPath -> runGhcid dotGhciPath Nothing
 
--- | Relative paths to local packages of an obelisk project
--- TODO a way to query this
-getLocalPkgs :: Applicative f => f [FilePath]
-getLocalPkgs = pure ["backend", "common", "frontend"]
+-- | Relative paths to local packages of an obelisk project.
+--
+-- These are a combination of the obelisk predefined local packages,
+-- and any packages that the user has set with the @packages@ argument
+-- to the Nix @project@ function.
+getLocalPkgs :: (MonadObelisk m, MonadIO m) => FilePath -> m [FilePath]
+getLocalPkgs root = do
+  root' <- liftIO $ makeAbsolute root
```

Comment:
Why not use `toNixPath` here since you went to the trouble of pulling out its logic into a named function?

Reply:
We need an absolute path here because nix always returns absolute paths for the project packages. I added a comment.